src/Runner.Common/JobServer.cs

```diff
@@ -56,6 +67,40 @@ public async Task ConnectAsync(VssConnection jobConnection)
             _hasConnection = true;
         }
 
+        private async Task CheckHealthEndpointsAsync()
+        {
+            try
```

Comment:
I would break these into 2 try-catch.
- try-catch _apis/health
- try-catch api.github.com
So we can see whether a particular service is having the issue.